src/internal/connector/onedrive/restore.go

```diff
@@ -217,129 +213,184 @@ func RestoreCollection(
 				continue
 			}
 
-			if backupVersion >= version.OneDrive1DataAndMetaFiles {
```

Comment:
This file got a refactoring so that I could have an easier time making assertions about the state of writes and counts.  Could still use a little more love, but now it should read more linearly for any given collection.